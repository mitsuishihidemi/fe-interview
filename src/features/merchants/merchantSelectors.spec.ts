import { getInitialState } from "../../tests/helpers";
import {
  selectMerchants,
  selectMerchantsFilteredByBill,
  selectShouldFetchMerchants,
  selectIsMerchantsPending,
} from "./merchantSelectors";

describe("selectMerchants", () => {
  it("should select merchants with category names", () => {
    const state = getInitialState();
    expect(selectMerchants(state)).toMatchSnapshot();
  });
});

describe("selectMerchantsFilteredByBill", () => {
  it("should select merchants filtered by isBill", () => {
    const state = getInitialState();
    const isBill = true;
    expect(selectMerchantsFilteredByBill(state, isBill)).toMatchSnapshot();
  });
});

describe("selectShouldFetchMerchants", () => {
  it("should return true if merchants status is idle", () => {
    const state = getInitialState({ merchants: { status: "idle" } });
    expect(selectShouldFetchMerchants(state)).toBe(true);
  });
  it("should return false if merchants status is not idle", () => {
    const state = getInitialState({ merchants: { status: "pending" } });
    expect(selectShouldFetchMerchants(state)).toBe(false);
  });
});

describe("selectIsMerchantsPending", () => {
  it("should return true if merchants status is pending", () => {
    const state = getInitialState({ merchants: { status: "pending" } });
    expect(selectIsMerchantsPending(state)).toBe(true);
  });

  it("should return false if merchants status is not pending", () => {
    const state = getInitialState({ merchants: { status: "idle" } });
    expect(selectIsMerchantsPending(state)).toBe(false);
  });
});
