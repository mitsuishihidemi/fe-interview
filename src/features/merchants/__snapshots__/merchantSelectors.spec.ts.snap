// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectMerchants should select merchants with category names 1`] = `
Array [
  Object {
    "category": "Phone",
    "categoryId": 1,
    "iconUrl": "https://pbs.twimg.com/profile_images/1151788824093188097/wHfb5mYZ_bigger.png",
    "id": "5a5caa1efe33900100fd8ed5",
    "isBill": true,
    "name": "Vodafone",
    "transactions": Array [
      Object {
        "amount": 12.34,
        "date": "2018-01-13",
        "id": 36,
      },
      Object {
        "amount": 14.34,
        "date": "2018-02-13",
        "id": 37,
      },
      Object {
        "amount": 15.54,
        "date": "2018-03-13",
        "id": 38,
      },
      Object {
        "amount": 11.34,
        "date": "2018-04-13",
        "id": 39,
      },
      Object {
        "amount": 18.99,
        "date": "2018-05-13",
        "id": 40,
      },
    ],
  },
  Object {
    "category": "Bank Charges",
    "categoryId": 3,
    "iconUrl": "https://pbs.twimg.com/profile_images/1280422740638470147/aBXwvFOs_bigger.jpg",
    "id": "5a5caad4fe33900100fd8ed7",
    "isBill": true,
    "name": "HSBC Mortgage",
    "transactions": Array [
      Object {
        "amount": 1023,
        "date": "2018-01-01",
        "id": 1,
      },
      Object {
        "amount": 1023,
        "date": "2018-02-01",
        "id": 2,
      },
      Object {
        "amount": 1023,
        "date": "2018-03-01",
        "id": 3,
      },
      Object {
        "amount": 1023,
        "date": "2018-04-01",
        "id": 4,
      },
      Object {
        "amount": 1023,
        "date": "2018-05-01",
        "id": 5,
      },
    ],
  },
  Object {
    "category": "TV",
    "categoryId": 2,
    "iconUrl": "https://pbs.twimg.com/profile_images/1235992718171467776/PaX2Bz1S_bigger.jpg",
    "id": "5a5cab18fe33900100fd8ed8",
    "isBill": false,
    "name": "Netflix",
    "transactions": Array [
      Object {
        "amount": 19,
        "date": "2018-01-02",
        "id": 6,
      },
      Object {
        "amount": 19,
        "date": "2018-02-02",
        "id": 7,
      },
      Object {
        "amount": 19,
        "date": "2018-03-02",
        "id": 8,
      },
      Object {
        "amount": 19,
        "date": "2018-04-02",
        "id": 9,
      },
      Object {
        "amount": 19,
        "date": "2018-05-02",
        "id": 10,
      },
    ],
  },
]
`;

exports[`selectMerchantsFilteredByBill should select merchants filtered by isBill 1`] = `
Array [
  Object {
    "category": "Phone",
    "categoryId": 1,
    "iconUrl": "https://pbs.twimg.com/profile_images/1151788824093188097/wHfb5mYZ_bigger.png",
    "id": "5a5caa1efe33900100fd8ed5",
    "isBill": true,
    "name": "Vodafone",
    "transactions": Array [
      Object {
        "amount": 12.34,
        "date": "2018-01-13",
        "id": 36,
      },
      Object {
        "amount": 14.34,
        "date": "2018-02-13",
        "id": 37,
      },
      Object {
        "amount": 15.54,
        "date": "2018-03-13",
        "id": 38,
      },
      Object {
        "amount": 11.34,
        "date": "2018-04-13",
        "id": 39,
      },
      Object {
        "amount": 18.99,
        "date": "2018-05-13",
        "id": 40,
      },
    ],
  },
  Object {
    "category": "Bank Charges",
    "categoryId": 3,
    "iconUrl": "https://pbs.twimg.com/profile_images/1280422740638470147/aBXwvFOs_bigger.jpg",
    "id": "5a5caad4fe33900100fd8ed7",
    "isBill": true,
    "name": "HSBC Mortgage",
    "transactions": Array [
      Object {
        "amount": 1023,
        "date": "2018-01-01",
        "id": 1,
      },
      Object {
        "amount": 1023,
        "date": "2018-02-01",
        "id": 2,
      },
      Object {
        "amount": 1023,
        "date": "2018-03-01",
        "id": 3,
      },
      Object {
        "amount": 1023,
        "date": "2018-04-01",
        "id": 4,
      },
      Object {
        "amount": 1023,
        "date": "2018-05-01",
        "id": 5,
      },
    ],
  },
]
`;
